modules/parent-join/src/main/java/org/elasticsearch/join/mapper/ParentIdFieldMapper.java

```diff
@@ -92,10 +92,10 @@ protected ParentIdFieldMapper(String name, boolean eagerGlobalOrdinals) {
 
     @Override
     protected void parseCreateField(ParseContext context) {
-        if (context.externalValueSet() == false) {
-            throw new IllegalStateException("external value not set");
-        }
-        String refId = (String) context.externalValue();
+        throw new UnsupportedOperationException("Cannot directly call parse() on a ParentIdFieldMapper");
+    }
+
+    public void indexValue(ParseContext context, String refId) {
```

Comment:
Small comment, maybe `createIdField` would be more precise, since we aren't actually indexing values here. A similar thought applies to `BinaryFieldMapper#indexValue`.

Reply:
We've been calling the equivalent methods on other field mappers (created for index-time scripts) `indexValue` - maybe we should change those as well?

Reply:
I would be in favor of that!